### PR TITLE
update code comments

### PR DESCRIPTION
**Pull request by chaochn47**
Update minor code comments to improve code readability when reviewing PR https://github.com/etcd-io/etcd/pull/15924

Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

### CODE REVIEW

File: server/storage/mvcc/hash.go [hunk 1]
```diff
@@ -86,7 +86,12 @@ type KeyValueHash struct {
 }
 
 type HashStorage interface {
-	// Hash computes the hash of the KV's backend.
+	// Hash computes the hash of the whole backend keyspace,
+	// including key, lease, and other buckets in storage.
+	// This is designed for testing ONLY!
```
Comment (fuweid) [inline server/storage/mvcc/hash.go:91]: > This is designed for testing ONLY!

It's align with proto definition. 

REF: https://github.com/etcd-io/etcd/blob/04346e870fc0823d12ca8220c3489635e9e15e72/api/etcdserverpb/rpc.proto#L224

**Review by fuweid — APPROVED** [review]
LGTM

**Review by jmhbnz — APPROVED** [review]
LGTM

**Review by ahrtr — APPROVED** [review]
LGTM

Thanks

**Pull request merged**